resources/frontend.rb

```diff
@@ -19,8 +19,18 @@
       variables['frontend'][new_resource.name] ||= {}
       variables['frontend'][new_resource.name]['default_backend'] ||= ''
       variables['frontend'][new_resource.name]['default_backend'] << new_resource.default_backend
-      variables['frontend'][new_resource.name]['bind'] ||= ''
-      variables['frontend'][new_resource.name]['bind'] << new_resource.bind
+      variables['frontend'][new_resource.name]['bind'] = []
+      if new_resource.bind.is_a? Hash
+        new_resource.bind.map do |addresses, ports|
+          addresses = Array(addresses) unless addresses.is_a? Array
+          ports = Array(ports) unless ports.is_a? Array
```

Comment:
When I have a moment I'll test, I have some impediment to work at the moment.
As far as I remember, there were problems managing various input types (string or arrays) and that control helps not to create arrays of array array etc ...